src/connectionManager.ts

```diff
@@ -11,46 +13,68 @@ export interface Connection {
 }
 
 export class ConnectionManager {
+  public static connectionManager: ConnectionManager;
   private connections: Record<string, Connection>;
   private currentConnection?: Connection;
+  private static _extContext: ExtensionContext;
 
-  constructor() {
+  private constructor() {
     this.connections = {};
   }
 
+  public static _initialize(ec: ExtensionContext) {
+    ConnectionManager.connectionManager = new ConnectionManager();
+    ConnectionManager.connectionManager.connections = {};
+    this._extContext = ec;
```

Comment:
If the connection manager will never be initialized, there should be some logic here to return a reference to the existing instance instead of re-initializing it when `_initialize` is called again.